include/ideep/operators/conv.hpp

```diff
@@ -1354,6 +1354,7 @@ struct convolution_forward
       const dims& dilates,
       const dims& padding_l,
       const dims& padding_r,
+      const size_t weights_hashkey = 0, /*this is to check inplace weight updates*/
```

Comment:
Why is the weight content related to the primitive cache logic?

Reply:
Sorry, I could have added a note in the PR about this. This is required to allow inplace weight updates for the cached pds. We have seen this with some of the kernels where they reorder the weights for efficient gemm compute and cache the reordered weights internally for the subsequent pds with the assumption that the weights don't change. This will fail if the cached pd comes with updated weights. That's why I had to hash the weights to decide whether to reuse or recreate the primitive.

Reply:
> We have seen this with some of the kernels where they reorder the weights for efficient gemm compute and cache the reordered weights internally for the subsequent pds with the assumption that the weights don't change

May I know what code you are talking about. Caching a reordered weight inside the kernel without knowing if it is a constant is by itself an issue. Also creating a hashkey with small portion of weight is not reliable.

Reply:
For automatic weight prepacking in PyTorch and IPEX, we would first make sure that the weight is a constant before packing it into an optimal layout, e.g., the JIT script graph is frozen. Usually that happens in the inference path.